### PR TITLE
fix: Improve Lineage chart UI -- follow-up 

### PR DESCRIPTION
**Pull request by ozandogrultan**
Signed-off-by: Ozan Dogrultan <ozan.dogrultan@deliveryhero.com>

### Summary of Changes

A follow-up to https://github.com/amundsen-io/amundsen/pull/1815, instead of assigning a fixed horizontal spacing between different levels of Lineage graph, this PR introduces the changes to set this spacing dynamically, based on the longest label length of that level.

Additionally, some changes are introduced to fix the transition effect of children nodes to move in and out from their respective parents. Previously, all nodes were transitioning from the root node.

### Tests

No test changes are needed for this UI change.

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by feng-tao** [comment]
cc @Golodhros for review

File: frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts [hunk 1]
```diff
@@ -252,9 +268,24 @@ export const compactLineage = (
...
+    (obj, item) => ({
+      ...obj,
+      [item.depth]: 0,
+    }),
+    { 0: 0 }
+  );
+  nodes.forEach((d, idx) => {
+    const nodeLabel = getNodeLabel(d, idx);
+    // Offset 10 pixels for each character
+    const currentNodeWidth = nodeLabel.length * 10 + NODE_RADIUS;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts:282]: I would do:
```suggestion
    const currentNodeWidth = nodeLabel.length * CHARACTER_OFFSET + NODE_RADIUS;
```

File: frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts [hunk 2]
```diff
@@ -252,9 +268,24 @@ export const compactLineage = (
...
+    if (currentNodeWidth > depthMaxNodeWidthMapping[d.depth]) {
+      depthMaxNodeWidthMapping[d.depth] = currentNodeWidth;
+    }
+  });
   return nodes.reduce((acc, n) => {
-    // Normalize nodes for fixed depth.
-    n.y = n.y < 0 ? n.depth * -NODE_WIDTH : n.depth * NODE_WIDTH;
+    n.y = getNodeWidth(n, depthMaxNodeWidthMapping);
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts:288]: Thanks, more clear!

File: frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts [hunk 3]
```diff
@@ -314,8 +345,11 @@ export const buildEdges = (g, targetNode, nodes) => {
     .enter()
     .insert('path', 'g')
     .attr('class', 'graph-link')
-    .attr('d', () => {
-      const o = { x: targetNode.x0 || 0, y: targetNode.y0 || 0 };
+    .attr('d', (d) => {
+      const o =
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts:349]: d is pretty standard and most will understand what it is, but I don't know what is 'o'. Maybe we can call it 'coordinates' or 'ubication' or something like that?

**Review by Golodhros — APPROVED** [review]
LGTM

Only some minor comments

**Review by Golodhros — APPROVED** [review]
LGTM

**Pull request merged**